packages/backend/src/functions/auth.ts

```diff
@@ -2,21 +2,32 @@ import * as functions from "firebase-functions"
 import { UserRecord } from "firebase-functions/v1/auth"
 import admin from "firebase-admin"
 import dotenv from "dotenv"
-import { GENERIC_ERRORS, logMsg } from "../lib/logs"
 import { getCurrentServerTimestampInMillis } from "../lib/utils"
-import { MsgType } from "../../types/enums"
+import { logAndThrowError, makeError, printLog, SPECIFIC_ERRORS } from "../lib/errors"
+import { LogLevel } from "../../types/enums"
 
 dotenv.config()
 
 /**
- * Auth-triggered function which writes a user document to Firestore.
+ * Record the authenticated user information inside the Firestore DB upon authentication.
+ * @dev the data is recorded in a new document in the `users` collection.
+ * @notice this method is automatically triggered upon user authentication in the Firebase app
+ * which uses the Firebase Authentication service.
  */
 export const registerAuthUser = functions.auth.user().onCreate(async (user: UserRecord) => {
     // Get DB.
     const firestore = admin.firestore()
 
     // Get user information.
-    if (!user.uid) logMsg(GENERIC_ERRORS.GENERR_NO_AUTH_USER_FOUND, MsgType.ERROR)
+    if (!user.uid) {
+        const error = SPECIFIC_ERRORS.SE_AUTH_NO_CURRENT_AUTH_USER
```

Comment:
we can maybe use `logAndThrowError` directly here?

Reply:
definitely